chrome/content/zotero/xpcom/style.js

```diff
@@ -190,7 +190,26 @@ Zotero.Styles = new function() {
 		}
 		
 		return _styles[id] || false;
-	}
+	};
+	
+	this.getAsync = Zotero.Promise.coroutine(function* (id, skipMappings, download=true) {
+		yield this.init();
+		var style = this.get(id, skipMappings);
+		
+		if (!style && download) {
+			try {
+				// Don't wait too long
+				Zotero.debug(`Fetching required style ${id} asynchronously`);
+				let styleText = yield Zotero.File.getContentsFromURLAsync(id, {timeout: 5000});
```

Comment:
> (The whole id situation in CSL is non-ideal — the id would ideally be non-changing, with a (changeable) rel="self" for download (in which case renamed-styles.json might still be used for creating server-side rel="link" redirects but wouldn't be necessary for client-side resolution. But that's a discussion for elsewhere.)

Yes, no disagreement there. There might be more old discussion on this, but this is what I could dig up:

settle on new base uri for styles, update ids accordingly
https://github.com/citation-style-language/styles/issues/2

settle on conventions for versioning
https://github.com/citation-style-language/styles/issues/7

Switch from "http" to "https" prefix for CSL style ID and "self"-links
http://xbiblio-devel.2463403.n2.nabble.com/Switch-from-quot-http-quot-to-quot-https-quot-prefix-for-CSL-style-ID-and-quot-self-quot-links-td7578982.html

call for comments on base URI issue
http://xbiblio-devel.2463403.n2.nabble.com/call-for-comments-on-base-URI-issue-td6097469.html